packages/ai/src/prompt/wrap-gateway-error.ts

```diff
@@ -1,21 +1,39 @@
-import {
-  GatewayAuthenticationError,
-  GatewayModelNotFoundError,
-} from '@ai-sdk/gateway';
+import { GatewayAuthenticationError } from '@ai-sdk/gateway';
 import { AISDKError } from '@ai-sdk/provider';
 
 export function wrapGatewayError(error: unknown): unknown {
-  if (
-    GatewayAuthenticationError.isInstance(error) ||
-    GatewayModelNotFoundError.isInstance(error)
-  ) {
+  if (!GatewayAuthenticationError.isInstance(error)) return error;
+
+  const isProductionEnv = process?.env.NODE_ENV === 'production';
+  const moreInfoURL = 'http://v6.ai-sdk.dev/unauthenticated-ai-gateway';
+
+  if (isProductionEnv) {
     return new AISDKError({
       name: 'GatewayError',
-      message:
-        'Unauthenticated. Configure AI_GATEWAY_API_KEY or configure and use a provider module. Learn more: https://vercel.link/unauthenticated-ai-gateway-v6',
-      cause: error,
+      message: `Unauthenticated. Configure AI_GATEWAY_API_KEY or configure and use a provider module. Learn more: ${moreInfoURL}`,
     });
   }
 
-  return error;
+  const devEnvironmentError = new AISDKError({
+    name: 'GatewayError',
+    message: `
+\u001b[1m\u001b[31mError: Unauthenticated request to AI Gateway.\u001b[0m
+
+To authenticate, set the \u001b[33mAI_GATEWAY_API_KEY\u001b[0m environment variable with your API key.
+
+Alternatively, you can configure and use a provider module instead of the AI Gateway.
+
+Learn more: \u001b[34m${moreInfoURL}\u001b[0m
+`,
+  });
+
+  // prune the stack to 5 lines for clarity
+  if (devEnvironmentError.stack) {
+    devEnvironmentError.stack = devEnvironmentError.stack
+      .split('\n')
+      .slice(0, 6)
+      .join('\n');
+  }
```

Comment:
hm how would this look like in practice? we do not know how deep the user side stack trace is.